libexec/bats-core/bats

```diff
@@ -80,6 +84,98 @@ expand_path() {
   printf -v "$result" '%s/%s' "$dirname" "${path##*/}"
 }
 
+# argument: exclude_literal
+is_bats_file() {
+  [[ "$1" == *".bats"* ]]
+}
+
+# argument suite_filename
+is_excluded() {
+  if [[ -z "$exclude_files" ]]; then
+    return 1
+  fi
+  
+  local suite_filename="$1"
+  read -ra EXCLUDE_FILES <<< "$exclude_files"
+
+  for exclude_file in "${EXCLUDE_FILES[@]}"
+  do
+    # 'file check' needed: $exclude_file can contain dir or file filter
+    if is_bats_file "$exclude_file"; then
```

Comment:
So what about the dirs?

Reply:
they should survive this exclusion, else the whole construct of a test-suite could be skipped according the given filter.
keep in mind: this method takes a specific suite_filename as argument (also documented above method).